src/librustdoc/html/format.rs

```diff
@@ -79,6 +79,14 @@ impl Buffer {
         self.buffer.push_str(s);
     }
 
+    crate fn insert_str(&mut self, idx: usize, s: &str) {
+        self.buffer.insert_str(idx, s);
+    }
+
+    crate fn push_str(&mut self, s: &str) {
```

Comment:
There’s write_str already

Reply:
This is intentional to keep the API both compatible with String and fmt::Formatter as we want to avoid noise in this PR at least.